index_alias_impl.go

```diff
@@ -184,7 +191,7 @@ func (i *indexAliasImpl) SearchInContext(ctx context.Context, req *SearchRequest
 	if req.PreSearchData != nil {
 		if requestHasKNN(req) {
```

Comment:
i dont think we should do this check here, since redistributePreSearchData() internally is supposed to redistribute knn/bm25/synonym specific pre search data based on the key set in req object.